Navigation Block: De-emphasize "Link" block.

jasmussen opened the pull request:
## Description

One thing recent rounds of FSE testing has surfaced, is that when you create a navigation menu and start empty, people seem to very commonly be choosing the "Link" block to insert their navigation links, when in nearly all cases the "Page Link" block would've been a better choice, actually suggesting pages on your site to link to.

This PR renames "Link" to "Custom Link", to de-emphasize it. Before:

<img width="558" alt="Screenshot 2021-03-24 at 09 37 30" src="https://user-images.githubusercontent.com/1204802/112280867-bc24cf00-8c85-11eb-833a-f7ed6c2eaa38.png">

After:

<img width="558" alt="Screenshot 2021-03-24 at 09 43 29" src="https://user-images.githubusercontent.com/1204802/112280879-be872900-8c85-11eb-981a-1cd0df29ed90.png">

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

jasmussen committed:
Navigation Block: De-emphasize "Link" block.

ellatrix reviewed (approved):
Makes sense. This is how it was done in the customiser as well if I remember correctly.

jasmussen committed:
Try: Fix tests.

jasmussen commented:
Thank you for the review!

Turns out there was a legitimate test failure, and in trying to fix it (see https://github.com/WordPress/gutenberg/pull/30172/commits/8f8b436c7d7bcd15700669355da859b6fef906e8), I touched code that @gwwar tinkered with just a few weeks ago. Because that specific test was sensitive, I've converted this PR to a draft until she has a chance to look and see if I fixed it right.

gwwar commented:
🤔 The current link search allows us to insert all types if we search for it (post, page, category, etc), did we want to limit it to an arbitrary title/url combo?

`packages/e2e-tests/specs/experiments/blocks/navigation.test.js` also needs to be updated similar to the navigation-editor e2e test.
`packages/block-library/src/navigation-link/test/hooks.js` needs updated snapshots.

eg `npm run test-unit packages/block-library/src/navigation-link/test/hooks.js --update-snapshot`

Let me know if you run into trouble with this.

annezazu reviewed (commented):
Agreed on the intent behind this and switching to Custom Link. Great thinking! Can't speak to the code so will leave that to others ;) Appreciate you jumping in on improving this so rapidly.

jasmussen committed:
Fix tests.

jasmussen commented:
> 🤔 The current link search allows us to insert all types if we search for it (post, page, category, etc), did we want to limit it to an arbitrary title/url combo?

Thanks for that clarification! Maybe? https://github.com/WordPress/gutenberg/issues/30170 was created to outline some of the confusion around the different types of blocks you can insert, and the fact that it all sort of blends together is part of the problem. However in absence of more obvious transformation tools, it's probably not a good time to make such a change. But it might be in the future.

Thanks a bunch for the test pointeres. I've udpated!

jasmussen commented:
~Question: should the newfangled responsive toggle default to on?~

Welp, I commented on the wrong PR. I meant to comment on https://github.com/WordPress/gutenberg/pull/30047

jasmussen committed:
Another test.
